### ISSUE TITLE
Out Of Memory - RAM

### ISSUE THREAD

**Issue by LeBrav** [notmahi/clip-fields#4]
Hi! I finished the installation, and wanted to try Training a CLIP-Field directly by doing:
`
python train.py dataset_path=nyu.r3d
`
I amb monitoring both RAM and GPUram. I see that when the code starts, data starts to load on RAM:
` 
Loading data: 100%|████████████████████████████████████████████████████████| 757/757 [00:05<00:00, 135.05it/s] ` 
` Upscaling depth and conf: 100%|████████████████████████████████████████████| 757/757 [00:04<00:00, 157.72it/s] ` 
` Calculating global XYZs: 100%|██████████████████████████████████████████████| 757/757 [00:14<00:00, 51.72it/s] 
` 
The previous code ocupies about 30GB of RAM. 
Then models such as Detic load on GPU, but when I arrive to line 177, /dataloaders/real_dataset.py my PC kills the process because of RAM OOM:
` # First, setup detic with the combined classes.
self._setup_detic_all_classes(view_data)`
**Why is data load on ram and not on GPU? Is there any way to lower the GB of memory used by using batches?**

**Comment by notmahi** [comment]
Hi, sorry for the delayed response.

Have you tried tuning the `subsample_prob` in the `DeticDenseLabelledDataset` class? In my experience it's better to label a lot of images with very low `subsample_prob` than the other way around.

**Comment by qiwen-1123** [comment]
Hi,
I just tried the `1-parse rgbd.ipynb`.
`dataset = R3DSemanticDataset(DATA_PATH, CUSTOM_LABELS)` this line of code takes about 30GB RAM, then the last line `torch.save(dataset, '../nyu_views.pth')` keeps taking RAM until it crashed. Is there any way to reduce the model size?

**Comment by LeBrav** [comment]
> Hi, I just tried the `1-parse rgbd.ipynb`. `dataset = R3DSemanticDataset(DATA_PATH, CUSTOM_LABELS)` this line of code takes about 30GB RAM, then the last line `torch.save(dataset, '../nyu_views.pth')` keeps taking RAM until it crashed. Is there any way to reduce the model size?

Hi, once you create R3DSemanticDataset, a subset of this will be selected. One way to reduce the dataet is to manually subsample the same dataset:
`            view_dataset = R3DSemanticDataset(cfg.maximg, cfg.dataset_path, cfg.custom_labels)
          
            indices_to_delete = torch.arange(1, len(view_dataset), cfg.sample_freq * 2)
            indices_to_delete = sorted(indices_to_delete, reverse=True)  # Reverse to delete from the end
                    
            
            view_dataset.global_xyzs = view_dataset.global_xyzs[::cfg.sample_freq]
            view_dataset.global_pcds = view_dataset.global_pcds[::cfg.sample_freq]
            view_dataset.poses = view_dataset.poses[::cfg.sample_freq]
            view_dataset._confidences = view_dataset._confidences[::cfg.sample_freq]
            view_dataset._depth_images = view_dataset._depth_images[::cfg.sample_freq]
            view_dataset.global_pcds = view_dataset.global_pcds[::cfg.sample_freq]
            view_dataset._reshaped_conf = view_dataset._reshaped_conf[::cfg.sample_freq]
            view_dataset._reshaped_depth = view_dataset._reshaped_depth[::cfg.sample_freq]
            view_dataset._rgb_images = view_dataset._rgb_images[::cfg.sample_freq]
            view_dataset.maximg = len(view_dataset._rgb_images)
`
Remember that what you are doing is saving the dataset variable into the disk, not the model.

**Comment by notmahi** [comment]
Hi @qiwen-1123 , did this resolve your memory problems? If so, I would like to close this issue.

**Comment by qiwen-1123** [comment]
Hi @notmahi, the above solution can ameliorate this problem to some extent. But when I try to record my own r3d data with Record3D, even in a small room, the r3d file is already larger than 300MB, then the program crashes in the line `dataset = R3DSemanticDataset(DATA_PATH, CUSTOM_LABELS)` because of RAM issue. Do you have any good suggestion for recording r3d data?

**Comment by qiwen-1123** [comment]
Hi, @notmahi, I just found that my fps setting is too high, which makes the data too large. Btw, I am also using stretch robot, I see you use clip-fields with stretch robot together, but I have problems in aligning robot map(NAV2) with clip-fields coordinates, do you mind I open another issue for this?

**Comment by notmahi** [comment]
Hi @qiwen-1123 , I was going to suggest lowering the fps but it seems like you found that option already.

Re: map alignment, during the original CLIP-Fields project, we did the map alignments manually for the experiments. I know [others are working](https://github.com/facebookresearch/home-robot) on improving the situation; however, I consider that outside of CLIP-Field's purview.

Thanks for your input to this work!
